1DIP/deriv2_cgls.ipynb

```diff
@@ -418,27 +516,42 @@
     }
    ],
    "source": [
-    "plt.plot(x)"
+    "#bop = VectorData(np.reshape(b,[-1]))\n",
+    "op = VectorData(bn)\n",
+    "cgls = CGLS(operator=Aop, data=bop, max_iteration=4, update_objective_interval=10)\n",
+    "cgls.run()\n",
+    "\n",
+    "plt.figure()\n",
+    "plt.plot(np.linspace(0,1,100),cgls.solution.as_array(), label='Least squares solution with early stopping')\n",
+    "plt.plot(np.linspace(0,1,100),x, label='Ground truth solution')\n",
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "90dcba3f",
+   "metadata": {},
+   "source": [
+    "We can also do Tikhonov regularisation using the CIL framework. For example, in the block framework below: "
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 122,
+   "execution_count": null,
    "id": "15ff9d79",
    "metadata": {},
    "outputs": [],
    "source": [
     "ig = Aop.domain_geometry()\n",
     "L = IdentityOperator(ig)\n",
-    "# L=np.identity(n)\n",
     "alpha = 0.01\n",
```

Comment:
This is a different value of the regularization parameter compared to what was used for the non-CIL tikhonov problem earlier in the notebook. Would be better to define alpha back then and reuse here to hopefully get the same solution without and with CIL and comment that CIL reproduces what was done without CIL.

Reply:
Thanks Jakob - have changed this and added a comparison plot between the CIL and numpy solutions at the end of the notebook.